Libraries/LibWeb/CredentialManagement/CredentialsContainer.cpp

```diff
@@ -17,25 +20,476 @@ GC::Ref<CredentialsContainer> CredentialsContainer::create(JS::Realm& realm)
+        //    internal method on origin, options, and sameOriginWithAncestors.
+        auto maybe_r = interfaces[0]->create(realm(), origin, options, same_origin_with_ancestors);
+        // If that threw an exception:
+        if (maybe_r.is_error()) {
+            // 1. Let e be the thrown exception.
+            auto e = maybe_r.error_value();
+            // 2. Queue a task on global’s DOM manipulation task source to run the following substeps:
+            queue_global_task(HTML::Task::Source::DOMManipulation, global, GC::create_function(realm().heap(), [&] {
+                HTML::TemporaryExecutionContext execution_context { realm(), HTML::TemporaryExecutionContext::CallbacksEnabled::Yes };
+
+                // 1. Reject p with e.
+                WebIDL::reject_promise(realm(), *promise, e);
+            }));
+            // 3. Terminate these substeps.
+            return;
+        }
+
+        auto r = maybe_r.release_value();
+
+        // 2. If r is a Credential or null, resolve p with r, and terminate these substeps.
+        if (r.has<Empty>()) {
+            WebIDL::resolve_promise(realm(), *promise, JS::js_null());
+            return;
+        }
+        if (r.has<GC::Ref<Credential>>()) {
+            auto& credential = r.get<GC::Ref<Credential>>();
+            WebIDL::resolve_promise(realm(), *promise, credential);
+            return;
+        }
+
+        // 3. Assert: r is an algorithm (as defined in §2.2.1.4 [[Create]] internal method).
+        VERIFY(r.has<GC::Ref<CreateCredentialAlgorithm>>());
+
+        // 4. Queue a task on global’s DOM manipulation task source to run the following substeps:
+        auto& r_algo = r.get<GC::Ref<CreateCredentialAlgorithm>>();
+        queue_global_task(HTML::Task::Source::DOMManipulation, global, GC::create_function(realm().heap(), [this, &global, promise = GC::Root(promise), r_algo = GC::Root(r_algo)] {
+            HTML::TemporaryExecutionContext execution_context { realm(), HTML::TemporaryExecutionContext::CallbacksEnabled::Yes };
+
+            // 1. Resolve p with the result of promise-calling r given global.
+            auto maybe_result = r_algo->function()(global);
+            if (maybe_result.is_error()) {
+                WebIDL::reject_promise(realm(), *promise, maybe_result.error_value());
+                return;
+            }
+
+            auto& result = maybe_result.value();
+            WebIDL::resolve_promise(realm(), *promise, result);
+        }));
+    }));
+
+    // 17. React to p:
+    auto on_completion = GC::create_function(realm().heap(), [&settings, type](JS::Value) -> WebIDL::ExceptionOr<JS::Value> {
+        // 1. Remove type from settings’ active credential types.
+        settings.active_credential_types().remove_first_matching([&](auto& v) { return v == type; });
+
+        return JS::js_undefined();
+    });
+    WebIDL::react_to_promise(*promise, on_completion, on_completion);
```

Comment:
same here. are we supposed to remove `type` from settings' active credential types on rejection of the promise?